examples/training/classification/densenet/imagenet/basic_training.py

```diff
@@ -0,0 +1,224 @@
+Author: [ianjjohnson](https://github.com/ianjjohnson)
+Date created: 2022/07/25
+Last modified: 2022/07/25
+Description: Use KerasCV to train a DenseNet using modern best practices for image classification
+"""
+
+"""
+## Overview
+KerasCV makes training state-of-the-art classification models easy by providing implementations of modern models, preprocessing techniques, and layers.
+In this tutorial, we walk through training a DenseNet model against the Imagenet dataset using Keras and KerasCV.
+This tutorial requires you to have KerasCV installed:
+```shell
+pip install keras-cv
+```
+"""
+
+"""
+## Setup and flags
+
+"""
+
+flags.DEFINE_string("imagenet_path", None, "Directory from which to load Imagenet.")
+flags.DEFINE_string(
+    "backup_path", None, "Directory which will be used for training backups"
+)
+flags.DEFINE_string(
+    "weights_path", None, "Directory which will be used to store weight checkpoints."
+)
+
+FLAGS = flags.FLAGS
+FLAGS(sys.argv)
+
+NUM_CLASSES = 1000
+BATCH_SIZE = 256
+IMAGE_SIZE = (300, 300)
+EPOCHS = 250
+
+"""
+## Data loading
+This guide uses the
+[Imagenet dataset](https://www.tensorflow.org/datasets/catalog/imagenet2012). Note that this requires manual download, and does not work out-of-the-box with TFDS.
+To get started, we first load the dataset from a command-line specified directory where ImageNet is stored as TFRecords.
+"""
+
+
+def parse_imagenet_example(example, IMAGE_SIZE):
+    # Read example
+    image_key = "image/encoded"
+    label_key = "image/class/label"
+    keys_to_features = {
+        image_key: tf.io.FixedLenFeature((), tf.string, ""),
+        label_key: tf.io.FixedLenFeature([], tf.int64, -1),
+    }
+    parsed = tf.io.parse_single_example(example, keys_to_features)
+
+    # Decode and resize image
+    image_bytes = tf.reshape(parsed[image_key], shape=[])
+    image = tf.io.decode_jpeg(image_bytes, channels=3)
+    image = tf.image.resize(image, IMAGE_SIZE)
+
+    # Decode label
+    label = tf.cast(tf.reshape(parsed[label_key], shape=()), dtype=tf.int32) - 1
+    label = tf.one_hot(label, NUM_CLASSES)
+    return image, label
+
+
+def load_imagenet_dataset():
+    train_filenames = [
+        f"{FLAGS.imagenet_path}/train-{i:05d}-of-01024" for i in range(0, 1024)
+    ]
+    validation_filenames = [
+        f"{FLAGS.imagenet_path}/validation-{i:05d}-of-00128" for i in range(0, 128)
+    ]
+
+    train_dataset = tf.data.TFRecordDataset(filenames=train_filenames)
+    validation_dataset = tf.data.TFRecordDataset(filenames=validation_filenames)
+
+    train_dataset = train_dataset.map(
+        lambda x: parse_imagenet_example(x, IMAGE_SIZE),
+        num_parallel_calls=tf.data.AUTOTUNE,
+    )
+    validation_dataset = validation_dataset.map(
+        lambda x: parse_imagenet_example(x, IMAGE_SIZE),
+        num_parallel_calls=tf.data.AUTOTUNE,
+    )
+
+    return train_dataset.batch(BATCH_SIZE), validation_dataset.batch(BATCH_SIZE)
+
+
+train, test = load_imagenet_dataset()
```

Comment:
eventually we will want to move this into a keras_cv.loader, but that can be a follow up PR.